packages/yoastseo/src/languageProcessing/researches/keyphraseDistribution.js

```diff
@@ -80,137 +92,253 @@ const maximizeSentenceScores = function( sentenceScores ) {
+				sentence = copySentences[ j ];
+				j++;
+			}
+
+			const mergedSentence = cloneDeep( copySentences[ i ] );
+			mergedSentence.text = mergedText;
+			mergedSentence.tokens = mergedTokens;
+			mergedSentence.sentenceParentNode = mergedParentNodes;
+			mergedSentence.sourceCodeRange.endOffset = endOffset;
+			mergedSentences.push( mergedSentence );
+			i = j;
+		} else {
+			mergedSentences.push( sentence );
+			i++;
+		}
+	}
+
+	return mergedSentences;
+};
 
 /**
  * Determines which portions of the text did not receive a lot of content words from keyphrase and synonyms.
  *
- * @param {Paper}       paper               The paper to check the keyphrase distribution for.
- * @param {Researcher}  researcher          The researcher to use for analysis.
+ * @param {Paper}       paper		The paper to check the keyphrase distribution for.
+ * @param {Researcher}  researcher	The researcher to use for analysis.
  *
- * @returns {Object} The scores of topic relevance per portion of text and an array of all word forms to highlight.
+ * @returns {KeyphraseDistributionResult} The scores of topic relevance per portion of text and an array of all word forms to highlight.
  */
 const keyphraseDistributionResearcher = function( paper, researcher ) {
 	const functionWords = researcher.getConfig( "functionWords" );
 	const matchWordCustomHelper = researcher.getHelper( "matchWordCustomHelper" );
 	const getContentWordsHelper = researcher.getHelper( "getContentWords" );
-	const wordsCharacterCount = researcher.getResearch( "wordsCharacterCount" );
-	const memoizedTokenizer = researcher.getHelper( "memoizedTokenizer" );
+	const wordsCharacterCount = researcher.getHelper( "wordsCharacterCount" );
+	const customSplitIntoTokensHelper = researcher.getHelper( "splitIntoTokensCustom" );
+	const customSentenceTokenizer = researcher.getHelper( "memoizedTokenizer" );
 
 	// Custom topic length criteria for languages that don't use the default value to determine whether a topic is long or short.
 	const topicLengthCriteria = researcher.getConfig( "topicLength" ).lengthCriteria;
+	let sentences = [];
+	if ( matchWordCustomHelper ) {
+		// This is currently only applicable for Japanese.
+		let text = paper.getText();
+		text = removeHtmlBlocks( text );
+		text = filterShortcodesFromHTML( text, paper._attributes && paper._attributes.shortcodes );
+		text = mergeListItems( text );
+		sentences = getSentences( text, customSentenceTokenizer );
+	} else {
+		sentences = getSentencesFromTree( paper.getTree(), true );
+		sentences = mergeListItemSentences( sentences );
+	}
 
-	let text = paper.getText();
-	text = removeHtmlBlocks( text );
-	text = filterShortcodesFromHTML( text, paper._attributes && paper._attributes.shortcodes );
-	text = mergeListItems( text );
-	const sentences = getSentences( text, memoizedTokenizer );
+	// When the custom helper is available, we're using the sentences retrieved from the text for the analysis.
```

Comment:
I think this comment should be above the `if ( matchWordCustomHelper ) {` line? Otherwise I'm not sure what it's referring to 😅